runtime.md

```diff
@@ -15,3 +15,20 @@ Runs a process in a container. Can be invoked several times.
 Not sure we need that from oc cli. Process is killed from the outside.
 
 This event needs to be captured by oc to run onstop event handlers.
+
+## Hooks
+Hooks allows one to run code before/after various lifecycle events of the container. The state of the container is passed to the hooks over a well-known fd, so the hooks could get the information they need to do their work.
+
```

Comment:
We can provide an example. However, does it make sense to standardize what all is passed?
I think that will be platform specific for sure but also implementation specific up-to some extent. 
Right now I can think of this at minimum for Linux, if we were to define it:
```
{
    "pid": 1234,
    "root": "/path/to/container/root",
    "namespaces" : [
       {
        "type": "process",
        "path": "/proc/1234/ns/pid"
       },
     ...
    ]
     
}

WDYT?